spec/requests/api/v2/providers/courses/sync_with_search_and_compare_spec.rb

```diff
@@ -93,18 +93,83 @@
         end
       end
 
-      context 'invalid enrichment' do
-        let(:invalid_enrichment) { create(:course_enrichment, :with_invalid_content) }
-
+      context 'fee type based course' do
         let(:course) { create(:course, :fee_type_based, provider: provider, enrichments: [invalid_enrichment]) }
-        it { should have_http_status(:unprocessable_entity) }
 
-        it 'has validation errors' do
-          expect(json_data.count).to eq 4
-          expect(json_data[0]["detail"]).to eq("Reduce the word count for about course")
-          expect(json_data[1]["detail"]).to eq("Reduce the word count for interview process")
-          expect(json_data[2]["detail"]).to eq("Reduce the word count for how school placements work")
-          expect(json_data[3]["detail"]).to eq("Reduce the word count for fee details")
+        context 'invalid enrichment with invalid content exceed word count fields' do
+          let(:invalid_enrichment) {
+            create(:course_enrichment, :with_fee_based_course,
+                   about_course:  Faker::Lorem.sentence(400 + 1),
+                             interview_process:  Faker::Lorem.sentence(250 + 1),
+                             qualifications:  Faker::Lorem.sentence(100 + 1),
+                             how_school_placements_work:  Faker::Lorem.sentence(350 + 1),
+                             fee_details:  Faker::Lorem.sentence(250 + 1),
+                             salary_details:  Faker::Lorem.sentence(250 + 1))
+          }
+
+          it { should have_http_status(:unprocessable_entity) }
+
+          it 'has validation errors' do
```

Comment:
I think triggering one of the validation errors and checking it shows up is enough, no need to check all of them, it feels a bit like over testing and might make changes more difficult. We just want to test that given an invalid object, we get the validation displayed in the JSON output. What do you think? Not a blocker or anything.

Reply:
The json output has the intention of being used on the frontend and displayed as is without translating it further like the site versions. Hence triggering it all as it is user facing. 

The enrichment model error message is prefix with `^`.

The course takes that message then transform it further and changing the error structure as the validation is only appropriate for the latest enrichment vs all enrichments.

Hence I over testing at its smallest and innerest level, and at the most outterest.

Reply:
I'm going to leave it be.
To show all expected validations messages.


ie 
`json_data[0]["detail"]`

how does that related to 
![image](https://user-images.githubusercontent.com/470137/57515364-5b4ab280-730a-11e9-8334-9e6876611574.png)


So it most likely subject to change